core/src/test/java/org/elasticsearch/monitor/fs/FsInfoTests.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.monitor.fs;
+
+import java.util.Arrays;
+
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class FsInfoTests extends ESTestCase {
+
+    public void testFsInfo() {
+        final FsInfo.Path[] paths = randomPaths();
+        final long timestamp = randomLong();
+        FsInfo fsInfo = new FsInfo(timestamp, null, paths);
+
+        long size = Arrays.stream(paths).map(n -> n.getTotal().getBytes()).mapToLong(n -> n).sum();
+        assertThat(timestamp, equalTo(fsInfo.getTimestamp()));
+        assertThat(size, equalTo(fsInfo.getTotal().getTotal().getBytes()));
+    }
+
+    public void testFsInfoSameMount() {
+        FsInfo.Path[] paths = randomPaths();
+        final String mount = randomAlphaOfLength(10);
+        Arrays.stream(paths).forEach(p -> p.mount = mount);
+        final long timestamp = randomLong();
+        FsInfo fsInfo = new FsInfo(timestamp, null, paths);
+
+        long size = paths[0].getTotal().getBytes();
+        assertThat(timestamp, equalTo(fsInfo.getTimestamp()));
+        assertThat(size, equalTo(fsInfo.getTotal().getTotal().getBytes()));
+    }
+
+    public void testFsInfoSameDevices() {
+        FsInfo.Path[] paths = randomPaths();
+        final String path = randomAlphaOfLength(10);
+        Arrays.stream(paths).forEach(p -> p.path = path);
+        final long timestamp = randomLong();
+        FsInfo fsInfo = new FsInfo(timestamp, null, paths);
+
+        long size = Arrays.stream(paths).map(n -> n.getTotal().getBytes()).mapToLong(n -> n).sum();
+        assertThat(timestamp, equalTo(fsInfo.getTimestamp()));
+        assertThat(size, equalTo(fsInfo.getTotal().getTotal().getBytes()));
+    }
+
+    public void testFsInfoNullMount() {
```

Comment:
The existing code doesn't filter out paths whose `mount` property is `null`, so I don't think we should start doing so as per this test case.